lib/client/mfa.go

```diff
@@ -110,19 +113,36 @@ func (tc *TeleportClient) PromptMFAChallenge(ctx context.Context, proxyAddr stri
 		applyOpts(opts)
 	}
 
+	// Default to local mfa challenge.
 	return promptMFAStandalone(ctx, c, addr, opts)
 }
 
 // PromptMFAChallenge prompts the user to complete MFA authentication
 // challenges.
 func PromptMFAChallenge(ctx context.Context, c *proto.MFAAuthenticateChallenge, proxyAddr string, opts *PromptMFAChallengeOpts) (*proto.MFAAuthenticateResponse, error) {
+	// First, check for a forwarded ssh agent that supports remote mfa challenges.
+	socketPath := os.Getenv(teleport.SSHAuthSock)
+	conn, err := agentconn.Dial(socketPath)
+	if err == nil {
+		defer conn.Close()
+		agentclient := agent.NewClient(conn)
+		supportedExtension, err := callQueryExtension(agentclient)
+		if err == nil && supportedExtension[promptMFAChallengeAgentExtension] {
+			resp, err := callPromptMFAChallengeExtension(agentclient, proxyAddr, c, opts)
+			if err == nil {
```

Comment:
Should we ignore this error? Previous ones makes sense, but here we're in territory "agent is forwarded, extension is supported", so feels like this error is now significant, maybe at least log it.